uvtools/dspec.py

```diff
@@ -1876,52 +1920,109 @@ def _fit_basis_2d(x, data, wgts, filter_centers, filter_half_widths,
         basis_options = [{k:basis_options[k][0] for k in basis_options}, {k:basis_options[k][1] for k in basis_options}]
     #filter -1 dimension
     model = np.zeros_like(data)
-    for i, _y, _w, in zip(range(data.shape[0]), data, wgts):
-        if np.count_nonzero(_w)/len(_w) >= skip_wgt and np.count_nonzero(_w[:max_contiguous_edge_flags]) > 0 \
-                                                        and np.count_nonzero(_w[-max_contiguous_edge_flags:]) >0:
-            model[i], _, info_t = _fit_basis_1d(x=x[1], y=_y, w=_w, filter_centers=filter_centers[1],
-                                            filter_half_widths=filter_half_widths[1],
-                                            suppression_factors=suppression_factors[1],
-                                            basis_options=basis_options[1], method=method,
-                                            basis=basis, cache=cache)
-            info['status']['axis_1'][i] = 'success'
-        else:
-            info['status']['axis_1'][i] = 'skipped'
-    #and if filter2d, filter the 0 dimension. Note that we feed in the 'model'
-    #set wgts for time filtering to happen on skipped rows
-    info['filter_params'] = {'axis_0':{}, 'axis_1':{}}
-    if np.any([info['status']['axis_1'][i] == 'success' for i in info['status']['axis_1']]):
-        info['filter_params']['axis_1']['method'] = info_t['method']
-        info['filter_params']['axis_1']['basis'] = info_t['basis']
-        info['filter_params']['axis_1']['filter_centers'] = info_t['filter_centers']
-        info['filter_params']['axis_1']['filter_half_widths'] = info_t['filter_half_widths']
-        info['filter_params']['axis_1']['suppression_factors'] = info_t['suppression_factors']
-        info['filter_params']['axis_1']['basis_options'] = info_t['basis_options']
-        info['filter_params']['axis_1']['mode'] = info_t['basis'] + '_' + method
-    if filter2d:
-        wgts_time = np.ones_like(wgts)
-        for i in range(data.shape[0]):
-            if info['status']['axis_1'][i] == 'skipped':
-                wgts_time[i] = 0.
-        for i, _y, _w, in zip(range(model.shape[1]), model.T, wgts_time.T):
+    if filt2d_mode == 'outer' or not filter2d:
+        for i, _y, _w, in zip(range(data.shape[0]), data, wgts):
```

Comment:
An explanatory comment would be helpful here (it's a little hard to parse whether the diff makes sense without knowing what this is meant to do).